homeassistant/components/openuv/config_flow.py

```diff
@@ -51,40 +78,96 @@ def config_schema(self) -> vol.Schema:
             }
         )
 
-    async def _show_form(self, errors: dict[str, Any] | None = None) -> FlowResult:
-        """Show the form to the user."""
-        return self.async_show_form(
-            step_id="user",
-            data_schema=self.config_schema,
-            errors=errors if errors else {},
-        )
+    async def _async_verify(
+        self, data: OpenUvData, error_step_id: str, error_schema: vol.Schema
+    ) -> FlowResult:
+        """Verify the credentials and create/re-auth the entry."""
+        websession = aiohttp_client.async_get_clientsession(self.hass)
+        client = Client(data.api_key, 0, 0, session=websession)
+        client.disable_request_retries()
+
+        try:
+            await client.uv_index()
+        except OpenUvError:
+            return self.async_show_form(
+                step_id=error_step_id,
+                data_schema=error_schema,
+                errors={CONF_API_KEY: "invalid_api_key"},
+                description_placeholders={
+                    CONF_LATITUDE: str(data.latitude),
+                    CONF_LONGITUDE: str(data.longitude),
+                },
+            )
+
+        entry_data = {
+            CONF_API_KEY: data.api_key,
+            CONF_LATITUDE: data.latitude,
+            CONF_LONGITUDE: data.longitude,
+            CONF_ELEVATION: data.elevation,
+        }
+
+        if existing_entry := await self.async_set_unique_id(data.unique_id):
+            self.hass.config_entries.async_update_entry(existing_entry, data=entry_data)
+            self.hass.async_create_task(
+                self.hass.config_entries.async_reload(existing_entry.entry_id)
+            )
+            return self.async_abort(reason="reauth_successful")
+        return self.async_create_entry(title=data.unique_id, data=entry_data)
```

Comment:
I think this isn't reachable for the reauth flow?  We don't want them to be able to change accounts/unique ids in the middle of a reauth.

Reply:
Not following. Agreed that this isn't reachable by reauth – we'd only reach this line if we created a new entry. What am I missing?

Reply:
Sorry I did a poor job of explaining (jet lag was kicking in before I went to go get dinner).

I wanted to make sure they could not end up with a new config entry as a result of the reauth flow.  Its a big cleaner to separate final function of the reauth flow so its clear that this can't happen but I can see that from the code that the net result of the reauth is never a new entry since `async_create_entry` isn't reachable from reauth.